src/model/test/GroundHeatExchangerHorizontalTrench_GTest.cpp

```diff
@@ -189,7 +189,7 @@ TEST_F(ModelFixture, GroundHeatExchangerHorizontalTrench_Clone) {
 
     auto ghClone = gh.clone(m).cast<GroundHeatExchangerHorizontalTrench>();
     EXPECT_EQ(2u, m.getConcreteModelObjects<GroundHeatExchangerHorizontalTrench>().size());
-    EXPECT_EQ(1u, m.getConcreteModelObjects<SiteGroundTemperatureUndisturbedKusudaAchenbach>().size());
+    EXPECT_EQ(2u, m.getConcreteModelObjects<SiteGroundTemperatureUndisturbedKusudaAchenbach>().size());
```

Comment:
Should clone into the same model create a new SGT object here? If not, how can this be prevented?